src/cbmc_viewer/coveraget.py

```diff
@@ -476,59 +476,56 @@ def update_coverage(coverage, path, func, line, status):
     return coverage
 
 ################################################################
-# Parse the basic block of code in a coverage goal
-
-def parse_coverage_description(string):
-    """Parse a coverage description and extract the basic block covered."""
-
-    # The coverage description has the form "block N (lines BASIC_BLOCK)"
-    match = re.match(r'block [0-9]+ \(lines (.*)\)', string)
-    if not match:
-        logging.info("Found unparsable basic block description: %s", string)
-        return None
-
-    return parse_basic_block(match.group(1))
-
-def parse_basic_block(basic_block):
-    """Parse a basic block of coverage"""
-
-    # The basic block is a sequence of chunks CHUNK1;CHUNK2;CHUNK3
-    return [location
-            for chunk in basic_block.split(';')
-            for location in parse_chunk(chunk)]
-
-def parse_chunk(chunk):
-    """Parse a chunk of coverage"""
-
-    # cbmc has added whitespace to coverage descriptions
-    chunk = chunk.strip()
-
-    # The chunk has the form FILE:FUNCTION:LINES
-    # Don't use split in order to allow FUNCTION to contain ':'
-    first_colon, last_colon = chunk.find(':'), chunk.rfind(':')
-    filename = chunk[:first_colon]
-    function = chunk[first_colon + 1:last_colon]
-    lines = chunk[last_colon + 1:]
-    return [(filename, function, line) for line in parse_lines(lines)]
+# Extract the source lines contributing to a coverage goal's basic block
+#
+# Each coverage goal is a basic block in a goto program. Each
+# instruction in a basic block has a source location identifying the
+# source line that generated the instruction.
+
+def parse_lines(string):
+    "Extract line numbers from the string encoding of line numbers used in coverage output"
+
+    # string is an encoding of a set of line numbers like
+    #   "1,3,6" -> {1, 3, 6}
+    #   "1-3,6,20-32" -> {1, 2, 3, 6, 20, 21, 22}
+
+    lines = set()
+    for group in string.split(','):
+        bounds = group.split('-')
+        if len(bounds) == 1:
+            lines.add(int(bounds[0]))
+            continue
+        if len(bounds) == 2:
+            lines.update(range(int(bounds[0]), int(bounds[1])+1))
+            continue
+        raise UserWarning(f"Unexpected encoding of line numbers: {string}")
```

Comment:
Nit picking: why not write this as `if ... elif ... else`?

Reply:
Done.